ci-setups/php-gitlab.md

```diff
@@ -0,0 +1,140 @@
+        # Run pipeline if a merge request is created for these branches
+        - if: $CI_MERGE_REQUEST_TARGET_BRANCH_NAME == "main" || $CI_MERGE_REQUEST_TARGET_BRANCH_NAME == "another-branch"
+      ```
+
+      This is evaluated as follows:
+      - If the commit branch is 'main' OR 'some-branch' OR 'some-other-branch', run the job.
+      - If a merge request is created and target branch is 'main' OR 'another-branch', run the job.
+
+    - ```yaml
+      - echo "Copy .env file"
+      - cp ${ENV_FILE} .env
+      ```
+
+      This can be removed if the project does not have or require any `.env` file for composer operations.
+
+      > Note: Here, the CI/CD Variable `ENV_FILE` is of type `FILE`. Depending on the type of Variable being used, this section can be edited as required.
+
+    - ```yaml
+      - echo "Updating dependencies"
+      - composer update
+      ```
+
+      This can be removed in case it is not desired to update the composer dependencies before executing commands.
+
+3. Add or edit the commit message, choose the main branch of your project and click on **Commit changes**. You should get a banner message like follows to signify that the file has been created successfully.
+
+    ![Gitlab Pipeline created successfully](assets/php-gitlab-pipeline-created.png)
+
+    The creation of this `.gitlab-ci.yml` file can also be verified by checking in the root of the project.
+
+#### Method 2: Manually creating pipeline file
+
+1. Create or clone the repository on your system.
+2. In the root of your project, create a file `.gitlab-ci.yml`.
+3. Add the same content as mentioned in step 2 of [Method 1: Using Gitlab UI](#method-1-using-gitlab-ui) above to this file. Make changes as required.
+4. Commit and push this file to the main branch of your repository. The workflow file should now be created in your project.
+
+### Verifying the pipeline
+
+After following the steps for creating a new pipeline, you can verify if the pipeline is running as expected or not. For this, follow the below mentioned steps:
+
+1. Open your project repository and click on **Build->Pipelines**. This should open a page as follows:
+
+    ![Gitlab Pipelines](assets/php-gitlab-pipelines.png)
+
+2. Click on any of the listed pipeline to view details about commit or merge request for which it was executed.
+
+    Click on the Pipeline `#<number>` to view further details about the pipeline such as the Jobs run. This can also be directly opened from the main Pipelines page.
+
+    ![Gitlab Pipeline for commit](assets/php-gitlab-pipeline-commit.png)
+
+3. Click on a Job to view complete details about where the execution has failed for a `failed` pipeline. This information can be used to perform required fixes.
+
+    ![Gitlab Pipeline jobs](assets/php-gitlab-pipeline-job.png)
+
+4. Upon performing the needed fixes, the pipeline  status should be `passed` after the jobs have been run.
+
+    ![Gitlab Pipeline passed](assets/php-gitlab-pipeline-passed.png)
```

Comment:
Can we also add 1 section to ask the maintainer to make the following changes.

1. PR check should pass before merging.
2. If CI check doe snot pass then we should not allow to merge
3. But should let some leads to bypass it (because there can be some cases where we want that to happen)

Reply:
This section has been added in the documentation restructuring